envbuilder.go

```diff
@@ -354,8 +383,10 @@ ENTRYPOINT [%q]`, exePath, exePath, exePath)
 	stderrWriter, closeStderr := log.Writer(opts.Logger)
 	defer closeStderr()
 	build := func() (v1.Image, error) {
-		_, err := opts.Filesystem.Stat(constants.MagicFile)
-		if err == nil && opts.SkipRebuild {
+		defer cleanupBuildContext()
```

Comment:
There's a defer above as well, will we end up logging errors the second time?

Reply:
Good point, I wrapped it in a `sync.Once`.